src/transaction/boot_cl.c

```diff
@@ -6720,6 +6723,255 @@ boot_define_view_synonym (void)
+    {
+      return error_code;
+    }
+
+  error_code = au_change_owner (class_mop, Au_dba_user);
+  if (error_code != NO_ERROR)
+    {
+      return error_code;
+    }
+
+  return NO_ERROR;
+}
+
+static int
+boot_define_view_user_objects (void)
+{
+  MOP class_mop;
+  COLUMN columns[] = {
+    // {"owner", "varchar(255)"},
+    {"object_name", "varchar(255)"},
+    {"object_type", "varchar(255)"},
+    {"generated", "varchar(1)"}
+  };
+
+  int num_cols = sizeof (columns) / sizeof (columns[0]);
+  int i;
+  char stmt[2048];
+  int error_code = NO_ERROR;
+
+  /* Initialization */
+  memset (stmt, '\0', sizeof (char) * 2048);
+
+  class_mop = db_create_vclass (CTV_USER_OBJECTS_NAME);
+  if (class_mop == NULL)
+    {
+      assert (er_errid () != NO_ERROR);
+      error_code = er_errid ();
+      return error_code;
+    }
+
+  for (i = 0; i < num_cols; i++)
+    {
+      error_code = db_add_attribute (class_mop, columns[i].name, columns[i].type, NULL);
+      if (error_code != NO_ERROR)
+	{
+	  return error_code;
+	}
+    }
+
+// *INDENT-OFF*
+  sprintf (stmt, "SELECT [object_name], [object_type], [generated] FROM ( "
+  SELECT_OBJECTS_QUERY
+  ") WHERE [owner] = CURRENT_USER");
+// *INDENT-ON*
+
+  error_code = db_add_query_spec (class_mop, stmt);
+  if (error_code != NO_ERROR)
+    {
+      return error_code;
+    }
+
+  error_code = au_change_owner (class_mop, Au_dba_user);
+  if (error_code != NO_ERROR)
+    {
+      return error_code;
+    }
+
+  error_code = au_grant (Au_public_user, class_mop, AU_SELECT, false);
+  if (error_code != NO_ERROR)
+    {
+      return error_code;
+    }
+
+  return NO_ERROR;
+}
+
+static int
+boot_define_view_all_objects (void)
+{
+  MOP class_mop;
+  COLUMN columns[] = {
+    {"owner", "varchar(255)"},
+    {"object_name", "varchar(255)"},
+    {"object_type", "varchar(255)"},
+    {"generated", "varchar(1)"}
+  };
+
+  int num_cols = sizeof (columns) / sizeof (columns[0]);
+  int i;
+  char stmt[4096];
+  int error_code = NO_ERROR;
+
+  /* Initialization */
+  memset (stmt, '\0', sizeof (char) * 4096);
+
+  class_mop = db_create_vclass (CTV_ALL_OBJECTS_NAME);
+  if (class_mop == NULL)
+    {
+      assert (er_errid () != NO_ERROR);
+      error_code = er_errid ();
+      return error_code;
+    }
+
+  for (i = 0; i < num_cols; i++)
+    {
+      error_code = db_add_attribute (class_mop, columns[i].name, columns[i].type, NULL);
+      if (error_code != NO_ERROR)
+	{
+	  return error_code;
+	}
+    }
+
+  // *INDENT-OFF*
+  sprintf (stmt, "SELECT [owner], [object_name], [object_type], [generated] FROM ( "
+    SELECT_OBJECTS_QUERY
+  ") ");
+  // *INDENT-ON*
+
+  error_code = db_add_query_spec (class_mop, stmt);
+  if (error_code != NO_ERROR)
+    {
+      return error_code;
+    }
+
+  error_code = au_change_owner (class_mop, Au_dba_user);
+  if (error_code != NO_ERROR)
+    {
+      return error_code;
+    }
+
+  error_code = au_grant (Au_public_user, class_mop, AU_SELECT, false);
```

Comment:
Every user will see the same query result from ALL_OBJECTS as that which DBA users see from DBA_OBJECTS, and DBA users will see the same result from ALL_OBJECTS and DBA_OBJECTS.
DBA_OBJECTS seems to be redundant. Isn't it?

Reply:
I also agree with hyunikn.

Reply:
I agree with your opinion. I realize that our catalogs don't need such a 'dba_' system table. I'm going to leave only all_objects table and check current user is dba group in the WHERE claus of the query spec. Also, objects that the current user can be accessed will be found by OR predicate in the WHERE conditin like the other system views.